config/hocon-mp/src/main/java/io/helidon/config/hocon/mp/HoconMpConfigSourceProvider.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.config.hocon.mp;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.LinkedList;
+import java.util.List;
+
+import io.helidon.config.ConfigException;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+
+/**
+ * HOCON/JSON config source provider for MicroProfile config that supports file {@code application.conf} or file
+ * {@code application.json}. This class should not be used directly - it is loaded automatically by Java service loader.
+ */
+public class HoconMpConfigSourceProvider implements ConfigSourceProvider {
+    private static final List<String> APPLICATION_CONFIG_NAMES = List.of("application.json", "application.conf");
```

Comment:
We only document adding `application.yaml` as a default config source in Helidon MP, this may introduce unexpected behavior (and backward incompatible behavior)

Reply:
Any suggestions on what approach we should use for this instead? I.e. I'm still vague on how we can identify a config file as the hocon/json config source for MP

Reply:
Would it be better to just search for any files in the class path that ends with *.conf or *.json. The only thing I worry about this is that it may include parsing includers as well if they are placed on the same path. Or should we search in a directory path called `hocon` and any files that ends with  `*.conf` or `*.json` will be parsed as a hocon or json configsource?

Reply:
Or will this have to tie up with the new MP meta-config refactoring that you suggested?

Reply:
To add more questions about MP meta-config refactoring if that is the way to go to add hocon/json config, what would be the difference of this with the changes that you've made to add MP meta-config support in https://github.com/oracle/helidon/pull/2767? I can see that MpConfigProviderResolver.java (particularly buildConfig()) MpConfigBuilder.java, and MpMetaConfig.java seems to have all the code that already supports the use of mp-meta-config.yaml/properties to process meta-config.

Reply:
For the first iteration, we will support programmatic setup of MP configuration (where you use the methods on config source to explicitly add HOCON/json files to your builder).
When we add support for HOCON/json to MP meta configuration, users will be able to use it to configure config sources.
I don't think we should do "magical" discovery of HOCON/json config sources. Also you cannot search classpath for all files.
The changes in #2767 were adding yaml explicitly, as yaml was on classpath of Helidon MP from the beginning as our supported configuration format, with `application.yaml` being one of the default config sources we look for.
We MAY decide to do a similar thing for `application.conf` or `application.json`, but that feels like a backward incompatible change in MP

Reply:
done/implemented